src/lib/snack-bar/snack-bar-ref.ts

```diff
@@ -11,18 +11,12 @@ import {Observable} from 'rxjs/Observable';
 import {Subject} from 'rxjs/Subject';
 import {MdSnackBarContainer} from './snack-bar-container';
 
-// TODO(josephperrott): Implement onAction observable.
-
 /**
  * Reference to a snack bar dispatched from the snack bar service.
  */
 export class MdSnackBarRef<T> {
-  private _instance: T;
-
   /** The instance of the component making up the content of the snack bar. */
-  get instance(): T {
-    return this._instance;
-  }
+  instance: T;
```

Comment:
I can't, because the instance gets assigned by the service right after the component is instantiated. This is why I had to remove the getter in the first place. The issue is that we need the snack bar ref to create the injector and we need the inejctor to create the component. Alternatively, I can make the injection tokens on the custom injector public, but that seems even hackier.